authorization/invitation/service/invitation_service_blackbox_test.go

```diff
@@ -459,3 +460,89 @@ func (s *invitationServiceBlackBoxTest) TestIssueSpaceInvite() {
+	user := s.Graph.CreateUser()
+	inv := s.Graph.CreateInvitation(team, user, true)
+
+	resourceID, err := s.Application.InvitationService().Accept(s.Ctx, user.IdentityID(), inv.Invitation().AcceptCode)
+	require.NoError(s.T(), err)
+
+	require.Equal(s.T(), team.ResourceID(), resourceID)
+
+	assocs, err := s.Application.Identities().FindIdentityMemberships(s.Ctx, user.IdentityID(), nil)
+	require.NoError(s.T(), err)
+
+	require.Len(s.T(), assocs, 1)
+
+	require.Equal(s.T(), team.TeamID(), *assocs[0].IdentityID)
+	require.True(s.T(), assocs[0].Member)
+	require.Empty(s.T(), assocs[0].Roles)
+}
+
+func (s *invitationServiceBlackBoxTest) TestAcceptTeamRoleInvitation() {
+	team := s.Graph.CreateTeam()
+	user := s.Graph.CreateUser()
+	teamRole := s.Graph.CreateRole(s.Graph.LoadResourceType(authorization.IdentityResourceTypeTeam))
+	inv := s.Graph.CreateInvitation(team, user, false, teamRole)
+
+	resourceID, err := s.Application.InvitationService().Accept(s.Ctx, user.IdentityID(), inv.Invitation().AcceptCode)
+	require.NoError(s.T(), err)
+
+	require.Equal(s.T(), team.ResourceID(), resourceID)
+
+	assocs, err := s.Application.IdentityRoleRepository().FindIdentityRolesForIdentity(s.Ctx, user.IdentityID(), nil)
+	require.NoError(s.T(), err)
+
+	require.Len(s.T(), assocs, 1)
+
+	require.Equal(s.T(), team.TeamID(), *assocs[0].IdentityID)
+	require.False(s.T(), assocs[0].Member)
+	require.Len(s.T(), assocs[0].Roles, 1)
+	require.Equal(s.T(), teamRole.Role().Name, assocs[0].Roles[0])
+}
+
+func (s *invitationServiceBlackBoxTest) TestAcceptSpaceInvitation() {
+	space := s.Graph.CreateSpace()
+	user := s.Graph.CreateUser()
+	spaceRole := s.Graph.CreateRole(s.Graph.LoadResourceType(authorization.ResourceTypeSpace))
+	inv := s.Graph.CreateInvitation(space, user, spaceRole)
+
+	resourceID, err := s.Application.InvitationService().Accept(s.Ctx, user.IdentityID(), inv.Invitation().AcceptCode)
+	require.NoError(s.T(), err)
+
+	require.Equal(s.T(), space.SpaceID(), resourceID)
+
+	roles, err := s.Application.IdentityRoleRepository().FindIdentityRolesForIdentity(s.Ctx, user.IdentityID(), nil)
+	require.NoError(s.T(), err)
+
+	require.Len(s.T(), roles, 1)
+	require.Equal(s.T(), space.SpaceID(), roles[0].ResourceID)
+	require.False(s.T(), roles[0].Member)
+	require.Len(s.T(), roles[0].Roles, 1)
+	require.Equal(s.T(), spaceRole.Role().Name, roles[0].Roles[0])
+}
+
+func (s *invitationServiceBlackBoxTest) TestAcceptFailsForIncorrectIdentity() {
+	space := s.Graph.CreateSpace()
+	user := s.Graph.CreateUser()
+	spaceRole := s.Graph.CreateRole(s.Graph.LoadResourceType(authorization.ResourceTypeSpace))
+	inv := s.Graph.CreateInvitation(space, user, spaceRole)
+
+	otherUser := s.Graph.CreateUser()
+
+	_, err := s.Application.InvitationService().Accept(s.Ctx, otherUser.IdentityID(), inv.Invitation().AcceptCode)
+	require.Error(s.T(), err)
+	require.IsType(s.T(), errors.NotFoundError{}, err)
+}
+
+func (s *invitationServiceBlackBoxTest) TestAcceptFailsForUnknownAcceptCode() {
```

Comment:
Could you please add a test for "Accept code already used" ?

Reply:
What should this test do?  Attempt to create a second invitation with the same accept code?

Reply:
Attempt to accept the same invitation code twice.

Reply:
Done